File: src/main/java/ru/reosfire/WMLib/Guis/Inventory/Components/IComponentCreator.java
package ru.reosfire.WMLib.Guis.Inventory.Components;

import ru.reosfire.WMLib.Guis.Inventory.Gui;
import ru.reosfire.WMLib.Yaml.Default.Gui.ComponentConfig;

public interface IComponentCreator
{
    GuiComponent Create(ComponentConfig config, Gui gui);
}

File: src/main/java/ru/reosfire/wmlib/guis/inventory/components/IComponentCreator.kt
package ru.reosfire.wmlib.guis.inventory.components

import ru.reosfire.wmlib.guis.inventory.Gui
import ru.reosfire.wmlib.yaml.common.gui.ComponentConfig

fun interface IComponentCreator {
    fun create(config: ComponentConfig, gui: Gui): GuiComponent
}